src/EditorFeatures/CSharpTest/ConvertNamespace/ConvertToFileScopedNamespaceAnalyzerTests.cs

```diff
@@ -391,6 +392,290 @@ class C
             }.RunAsync();
         }
 
+        [Fact]
+        public async Task TestConvertToFileScopedWithDocComment()
+        {
+            await new VerifyCS.Test
+            {
+                TestCode = @"
+[|namespace N|]
+{
+    /// <summary/>
+    class C
+    {
+    }
+}
+",
+                FixedCode = @"
+namespace $$N;
+
+/// <summary/>
+class C
+{
+}
+",
+                LanguageVersion = LanguageVersion.CSharp10,
+                Options =
+                {
+                    { CSharpCodeStyleOptions.NamespaceDeclarations, NamespaceDeclarationPreference.FileScoped }
+                }
+            }.RunAsync();
+        }
+
+        [Fact]
+        public async Task TestConvertToFileScopedWithPPDirective1()
+        {
+            await new VerifyCS.Test
+            {
+                TestCode = @"
+[|namespace N|]
+{
+#if X
+    class C
+    {
+    }
+#else
+    class C
+    {
+    }
+#endif
+}
+",
+                FixedCode = @"
+namespace $$N;
+
+#if X
+class C
+{
+}
+#else
+class C
+{
+}
+#endif
```

Comment:
We could argue that we should not touch the disabled section. there are pros/cons each way.